src/core/thread/mesh_forwarder.cpp

```diff
@@ -907,7 +912,7 @@ Neighbor *MeshForwarder::UpdateNeighborOnSentFrame(Mac::TxFrame &aFrame, Error a
 #endif // OPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE
 
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
-    if ((aFrame.GetHeaderIe(Mac::CslIe::kHeaderIeId) != nullptr) && aFrame.IsDataRequestCommand())
+    if ((aFrame.GetHeaderIe(Mac::CslIe::kHeaderIeId) != nullptr) && aIsDataPoll)
```

Comment:
Thanks for catching this issue! With IEEE 802.15.4-2006, the MAC Command field was not encrypted. However, with IEEE 802.15.4-2015 and use of Information Elements, the MAC Command field is now encrypted.